File: Emulator/src/main/content/global/skill/slayer/npc/KuraskNPC.java
package content.global.skill.slayer.npc;

import content.global.skill.slayer.SlayerUtils;
import core.game.node.entity.combat.BattleState;
import core.game.node.entity.npc.AbstractNPC;
import core.game.node.entity.player.Player;
import core.game.world.map.Location;
import core.plugin.Initializable;
import org.rs.consts.NPCs;

/**
 * The type Kurask npc.
 */
@Initializable
public class KuraskNPC extends AbstractNPC {

    /**
     * Instantiates a new Kurask npc.
     *
     * @param id       the id
     * @param location the location
     */
    public KuraskNPC(int id, Location location) {
        super(id, location);
    }

    /**
     * Instantiates a new Kurask npc.
     */
    public KuraskNPC() {
        super(0, null);
    }

    @Override
    public AbstractNPC construct(int id, Location location, Object... objects) {
        return new KuraskNPC(id, location);
    }

    @Override
    public void checkImpact(BattleState state) {
        super.checkImpact(state);
        boolean effective = false;
        if (state.getAttacker() instanceof Player) {
            Player player = (Player) state.getAttacker();
            effective = SlayerUtils.hasBroadWeaponEquipped(player, state);
        }
        if (!effective) {
            state.setEstimatedHit(0);
            if (state.getSecondaryHit() > 0) {
                state.setSecondaryHit(0);
            }
        }
    }

    @Override
    public int[] getIds() {
        return new int[] { NPCs.KURASK_1608, NPCs.KURASK_1609, NPCs.KURASK_4229 };
    }
}

File: Server/src/main/content/global/skill/slayer/npc/KuraskNPC.kt
package content.global.skill.slayer.npc

import content.global.skill.slayer.SlayerUtils.hasBroadWeaponEquipped
import content.global.skill.slayer.Tasks
import core.game.node.entity.combat.BattleState
import core.game.node.entity.npc.AbstractNPC
import core.game.node.entity.player.Player
import core.game.world.map.Location
import core.plugin.Initializable

/**
 * The type Kurask npc.
 */
@Initializable
class KuraskNPC : AbstractNPC {
    /**
     * Instantiates a new Kurask npc.
     *
     * @param id       the id
     * @param location the location
     */
    constructor(id: Int, location: Location?) : super(id, location)

    /**
     * Instantiates a new Kurask npc.
     */
    constructor() : super(0, null)

    override fun construct(id: Int, location: Location, vararg objects: Any): AbstractNPC {
        return KuraskNPC(id, location)
    }

    override fun checkImpact(state: BattleState) {
        super.checkImpact(state)
        var effective = false
        if (state.attacker is Player) {
            val player = state.attacker as Player
            effective = hasBroadWeaponEquipped(player, state)
        }
        if (!effective) {
            state.estimatedHit = 0
            if (state.secondaryHit > 0) {
                state.secondaryHit = 0
            }
        }
    }

    override fun getIds(): IntArray {
        return Tasks.KURASKS.npcs
    }
}
